hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyRackFaultTolerant.java

```diff
@@ -192,11 +192,23 @@ private void chooseEvenlyFromRemainingRacks(Node writer,
       } finally {
         excludedNodes.addAll(newExcludeNodes);
       }
+      if (numResultsOflastChoose == results.size()) {
+        Map<String, Integer> nodesPerRack = new HashMap<>();
+        for (DatanodeStorageInfo dsInfo : results) {
+          String rackName = dsInfo.getDatanodeDescriptor().getNetworkLocation();
+          nodesPerRack.merge(rackName, 1, Integer::sum);
+        }
+        for (int numNodes : nodesPerRack.values()) {
+          if (numNodes > bestEffortMaxNodesPerRack) {
+            bestEffortMaxNodesPerRack = numNodes;
+          }
```

Comment:
Line 201~203 get max value, maybe we can use the following code to make it more clear and concise.
```java
        bestEffortMaxNodesPerRack =
            Math.max(bestEffortMaxNodesPerRack, Collections.max(nodesPerRack.values()));
```

Reply:
Thank you for your feedback. I'll prioritize implementing the required code changes promptly.